src/ipa/rpi/common/ipa_base.cpp

```diff
@@ -1504,14 +1553,21 @@ void IpaBase::applyFrameDurations(Duration minFrameDuration, Duration maxFrameDu
 	 * value possible.
 	 */
 	Duration maxShutter = Duration::max();
-	helper_->getBlanking(maxShutter, minFrameDuration_, maxFrameDuration_);
+	auto [vblank, hblank] = helper_->getBlanking(maxShutter, minFrameDuration_, maxFrameDuration_);
 
 	RPiController::AgcAlgorithm *agc = dynamic_cast<RPiController::AgcAlgorithm *>(
 		controller_.getAlgorithm("agc"));
 	agc->setMaxShutter(maxShutter);
+
+	RPiController::SyncAlgorithm *sync = dynamic_cast<RPiController::SyncAlgorithm *>(
+		controller_.getAlgorithm("sync"));
+	if (sync) {
+		LOG(IPARPI, Debug) << "setting sync frame duration to  " << (mode_.height + vblank) * ((mode_.width + hblank) * 1.0s / mode_.pixelRate);
+		sync->setFrameDuration((mode_.height + vblank) * ((mode_.width + hblank) * 1.0s / mode_.pixelRate));
+	}
```

Comment:
Maybe use a temporary for the ``(mode_.height + vblank) * ((mode_.width + hblank) * 1.0s / mode_.pixelRate)`` calc so it does not get done twice?